ivy_models/googlenet/googlenet.py

```diff
@@ -0,0 +1,115 @@
+# global
+import ivy
+import ivy_models
+from .layers import ConvBlock, Inception, Auxiliary
+
+
+class GoogLeNet(ivy.Module):
+    """
+    Inception-V1 (GoogLeNet) architecture.
+
+    Args::
+        num_classes (int): Number of output classes. Defaults to 1000.
+        base_width (int): The base width of the ResNet. Defaults to 64.
+        v (ivy.Container): Unused parameter. Can be ignored.
+
+    """
+
+    def __init__(
+        self,
+        num_classes=1000,
+        v: ivy.Container = None,
+    ):
+        if v is not None:
+            self.v = v
+        self.num_classes = num_classes
+        super(GoogLeNet, self).__init__(v=v)
+
+    def _build(self, *args, **kwargs):
+        self.conv1 = ConvBlock(3, 64, [7, 7], 2, padding="SAME")
+        self.pool1 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.conv2 = ConvBlock(64, 64, [1, 1], 1, padding="SAME")
+        self.conv3 = ConvBlock(64, 192, [3, 3], 1, padding="SAME")
+        self.pool3 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.inception3A = Inception(192, 64, 96, 128, 16, 32, 32)
+        self.inception3B = Inception(256, 128, 128, 192, 32, 96, 64)
+        self.pool4 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.inception4A = Inception(480, 192, 96, 208, 16, 48, 64)
+
+        self.aux4A = Auxiliary(512, self.num_classes)
+
+        self.inception4B = Inception(512, 160, 112, 224, 24, 64, 64)
+        self.inception4C = Inception(512, 128, 128, 256, 24, 64, 64)
+        self.inception4D = Inception(512, 112, 144, 288, 32, 64, 64)
+
+        self.aux4D = Auxiliary(528, self.num_classes)
+
+        self.inception4E = Inception(528, 256, 160, 320, 32, 128, 128)
+        self.pool5 = ivy.MaxPool2D([3, 3], 2, "SAME")
+
+        self.inception5A = Inception(832, 256, 160, 320, 32, 128, 128)
+        self.inception5B = Inception(832, 384, 192, 384, 48, 128, 128)
+        self.pool6 = ivy.AdaptiveAvgPool2d([1, 1])
+
+        self.dropout = ivy.Dropout(0.4)
+        self.fc = ivy.Linear(1024, self.num_classes)
+
+    def _forward(self, x):
```

Comment:
once we have the dual format support, the forward should have an extra condition block, please ref alexnet model